### ISSUE TITLE
training one subject, then training anothjer one on same model while preserving identity of both

### ISSUE THREAD

**Issue by 1blackbar** [TheLastBen/fast-stable-diffusion#291]
is that even possible?
I trained a woman for 3k and 30 pics, came out great, so  and it gave me ckpt that i used to train a man as new subject, but in the end the woman starts to look like that new subject - a man, this is really an issue to train multiple people with same model, so the only way is to do it all of them at once ? How to preserve the prrevious subjects when training new ones ?

**Comment by TheLastBen** [comment]
keep a few instance images of the first person in the instance folder.

**Comment by 1blackbar** [comment]
ok, trained 2 subjects, now put back 3  head shot images of 2 previous subjects and training 3rd subject on new 30 imgs , we will see how it will come out

**Comment by 1blackbar** [comment]
OK i think it works but it overfits previous subjects with 3 images and i kinda want to freeze their training...

**Comment by TheLastBen** [comment]
For example, if you're training a subject with 15 pics, train 200-300 steps with the " --train_text_encoder " argument, the resume the rest of the training without it (3000 steps), you'll get better style transfer and less overfitting. I'll add tat option shortly

**Comment by 1blackbar** [comment]
i resumed training and deleted 2 pics of first subjects  so theres just 2 pics per each, this is not enough it starts to lose identity when training even 500 steps and their identity was fine without additional training, so each new training starts to los eidentity , ill try without text encoder.
What im tryin to do really is to "fix " some subjects like actors , and train them for awhile , i thought you cn just train one actor for like 3k steps then start next actor with 3k steps and youll get two nicely trained subjects but with each training the whole latent space is leaking with new subject... is that even solvable ?

**Comment by TheLastBen** [comment]
you know that the dreambooth script has a problem of bleeding meaning every time you train a subject, it bleeds to others in the same class

**Comment by 1blackbar** [comment]
yes i know so there is no remedy? or ability to tone down trainig for subjects that were already trained ?

**Comment by TheLastBen** [comment]
cutting the text encoder training steps short helps

**Comment by 1blackbar** [comment]
ok i just ran atest - 16 subjects, 4 images per subject, 5000 steps, its gonna be a shitshow or its gonna work just fine, if not, ill try with text encoder disabling

**Comment by TheLastBen** [comment]
make sure you run with the text encoder 200 steps per subject

**Comment by 1blackbar** [comment]
where do i change that? im on defaults but im 50% into training so i can try it when this one finishes

**Comment by TheLastBen** [comment]
it trains randomly, you run the training for 200x16 with the text encoder, then remove the argument --train_text_encoder, after that resume training for the rest of the steps

**Comment by 1blackbar** [comment]
Ok some subjects overfit, some wherent there yet, theres morevariables to it, the ones that were already in SD overfit faster cause of resemblance, and the ones that are new barely pickedup likeness, theres just no way to get reasonably good results with so many moving parts out of our control, i did it as a test what would happen and its pretty mucjh what i got from other multimodel trainings, you can do it with 2 mayb 3 subjects but its still a risk one wont be there yet.
Until theres no way to stop other subjects from training , its just not going to work that well with like 10 people at once, but id be happy to be proven wrong.
Maybe loss detection and extra code to skip the ones that have already good likeness would help, but thats just theory.

**Comment by TheLastBen** [comment]
in a few minutes there will be a feature to limit the text_encoder aggressive training

**Comment by 1blackbar** [comment]
great , i also suspect that 2e is too high to train multiple subjects with 3000 steps  they pretty much are on border of overfitting and that rate it gets great likeness but also suffers when stylising into non photo related modes like comics , but thats easy to change by myself

**Comment by TheLastBen** [comment]
added, use https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/fast-DreamBooth.ipynb
Start with 10% and avoid to many subjects, keep it below 3.
